### DISCUSSION TITLE
Jackson 3.0, JSTEP-9: deprecation and removal of `jackson-datatype-hibernate` modules

### DISCUSSION THREAD

**Discussion by cowtowncoder** [FasterXML/jackson-future-ideas#76] [Polls]
(part of #72 uber-Discussion)

As per more general [JSTEP-9](../wiki/JSTEP-9) -- deprecation and removal of some Jackson 2.x modules from 3.0 -- the current is for Hibernate datatype modules to be discontinued, not upgrade or maintained for Jackson 3.0.
This is strictly based on years-long lack of active Maintainer for modules.

If an active maintainer was found, we would change the plan.

This Discussion is created just to gauge interest in having these modules available for Jackson 3.x.

NOTE: "removal from 3.0" simply means no migration to build 3.x versions -- not deletion of code or repo or anything like that. Modules will be supported for 2.x regardless.

**Comment by pjfanning** [comment]
Let's remove the modules in v3 and if there is later demand for it and someone volunteers to maintain it, then we can resurrect the modules. I might even volunteer myself. But at the start, we don't need to be proactive - we can react if people come to us to ask for it to be migrated to jackson 3.

**Reply by JooHyukKim** [reply]
I concur. Sounds like a cost-effective option with little to lose.

**Reply by cowtowncoder** [reply]
This has been my thinking as well, but wanted to hear what others think.

And on if to upgrade -- I would specifically only do that if there is someone willing to take some form of ownership: I can (and am willing to) help make mechanical upgrade, but I don't have Hibernate knowledge to work with actual functionality (nor time + interest to obtain enough).

**Comment by cowtowncoder** [comment]
After 1 month of public opinion poll, let's close this discussion.

Result: Hibernate module will not be converted/maintained with Jackson 3.0, unless new active maintainer is found.
As things are, this means only 2.x versions will be available.

**Reply by JooHyukKim** [reply]
Added small update on [JSTEP-9 Wiki](https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-9/_compare/c17b1fc3a40ddf9e9f4e958840c41fb2d5135b81...85606854b28505f1e3b471688579b1025e6e3a9c)

**Comment by cowtowncoder** [comment]
NOTE: Decision was overturned in November 2025 as a maintainer was found (@pjfanning stepping up!), and modules have 3.0.x versions, plus will be part of regular releases since 3.0.3.
